src/generator/arm_pollers.ts

```diff
@@ -123,18 +90,26 @@ export async function generateARMPollers(session: Session<CodeModel>): Promise<s
     const pollerName = camelCase(poller.name);
     let responseType = 'HTTPResponse';
     // HTTP Pollers do not need to perform the final get request since they do not return a model
-    let finalResponseDeclaration = 'FinalResponse() *http.Response';
+    let finalResponseDeclaration = 'FinalResponse(ctx context.Context) (*http.Response, error)';
     let finalResponse = `${finalResponseDeclaration} {
-        return p.pt.latestResponse().Response;
+        return p.pt.FinalResponse(ctx, p.pipeline, nil)
       }`;
     let pollUntilDoneResponse = '(*http.Response, error)';
     let pollUntilDoneReturn = 'p.FinalResponse(), nil';
+    let pollUntilDone = `return p.pt.PollUntilDone(ctx, frequency, p.pipeline, nil)`;
     let handleResponse = '';
     const schemaResponse = <SchemaResponse>poller.op.responses![0];
     let unmarshalResponse = 'nil';
     let pagerFields = '';
-    let finalResponseCheckNeeded = false;
     if (isPageableOperation(poller.op)) {
+      function finalPagerProcessing(name: string, params: string): string {
+        return `s := &${camelCase(responseType)}{}
```

Comment:
You'll want to use more than one letter here or you run the risk of it colliding with the receiver var which is always one letter (the first letter of the type).

Reply:
I'm thinking of something like `temp` or maybe `respType` like the method param. Any preference?

Reply:
respType sounds good to me

Reply:
done